### PR TITLE
(PC-18121)[PRO] feat: input time v3

### PR DESCRIPTION
**Pull request by hugocapuano**
Lien vers le ticket Jira : https://passculture.atlassian.net/browse/PC-18121

## But de la pull request

Reprendre (pour la v3) le fonctionnement du champ "durée" de la v2.

## Implémentation

- Création d'un composant DurationInput
- Tests associés au composant
- Suppression et modifications des tests pour OfferIndividualForm/Informations

### CODE REVIEW

File: pro/src/components/OfferIndividualForm/Informations/Informations.tsx [hunk 1]
```diff
@@ -99,11 +101,11 @@ const Informations = ({
 
       {hasDurationMinutes && (
         <FormLayout.Row>
-          <TextInput
+          <DurationInput
             isOptional
             label={'Durée'}
             name="durationMinutes"
-            placeholder="HH:MM"
+            className={styles['input-durationMinutes']}
```
Comment (Aliochka) [inline pro/src/components/OfferIndividualForm/Informations/Informations.tsx:108]: détail: je trouve ça étrange de mixer camel et kebab case comme c'est une classe css `input-duration-minutes` ?

File: pro/src/ui-kit/form/DurationInput/DurationInput.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  const onDurationBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    const [updatedHours, updatedMinutes] = event.target.value.split(':')
+    const updatedDurationInMinutes =
+      parseInt(updatedHours || '0') * 60 + parseInt(updatedMinutes || '0')
+    const durationMinutes = parseMinutesToHours(updatedDurationInMinutes)
+
+    helpers.setValue(durationMinutes)
+  }
+
+  const onDurationChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (Aliochka) [inline pro/src/ui-kit/form/DurationInput/DurationInput.tsx:33]: détail je les appelerais juste 
`onBlur` & `onChange`

File: pro/src/ui-kit/form/DurationInput/__specs__/DurationInput.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    { value: '2', expectedDuration: '2:00' },
+    { value: '02', expectedDuration: '2:00' },
+    { value: '2:', expectedDuration: '2:00' },
+    { value: '2:3', expectedDuration: '2:03' },
+    { value: '2:30', expectedDuration: '2:30' },
+    { value: ':30', expectedDuration: '0:30' },
+    { value: ':3', expectedDuration: '0:03' },
+    { value: 'A', expectedDuration: '' },
+  ]
+  it.each(setDurationInputValue)(
```
Comment (Aliochka) [inline pro/src/ui-kit/form/DurationInput/__specs__/DurationInput.spec.tsx:43]: c'est vraiment cool les test paramétrés comme ça je trouve !

sinon un edge case potentiellement relou 10e2 est un nombre en js, je me demande s'il y en a d'autres

**Review by Aliochka — APPROVED** [review]
top !

si les ptits commit de fix concernent ce que t'as fait dans le même ticket tu peux les squasher dans le commit qu'ils complètent

File: pro/src/ui-kit/form/DurationInput/__specs__/DurationInput.spec.tsx (hunk 3 above)
Comment (hugocapuano) [reply pro/src/ui-kit/form/DurationInput/__specs__/DurationInput.spec.tsx:43]: Je ne sais pas qui fait ça dans le projet, mais j'ai repris l'idée dans d'autres tests du projet ! super pratique :)
Concernant 10e2 ça ne va pas fonctionner, même avec un copier/coller, il n'y aura que 10 qui va s'écrire, les lettres ne modifient pas la valeur du champ

File: pro/src/ui-kit/form/DurationInput/DurationInput.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    ),
+})
+
+const Template: Story<Args> = args => (
+  <Formik
+    initialValues={args.initialValues}
+    validationSchema={validationSchema}
+    onSubmit={() => {}}
+  >
+    <DurationInput {...args} />
```
Comment (rlecellier) [inline pro/src/ui-kit/form/DurationInput/DurationInput.stories.tsx:38]: étrange que `DurationInput` ne rale pas alors que tu lui donne une prop inconnu `initialValues`.

File: pro/src/ui-kit/form/DurationInput/DurationInput.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+interface Args extends IDurationInputProps {
+  initialValues: { durationMinutes: string | null }
+}
+
+const args = {
+  initialValues: { durationMinutes: '' },
+  name: 'durationMinutes',
+}
+
+const validationSchema = yup.object().shape({
```
Comment (rlecellier) [inline pro/src/ui-kit/form/DurationInput/DurationInput.stories.tsx:22]: Ce schema pourrait être dans le dossier du composant `DurationInput`.

File: pro/src/ui-kit/form/DurationInput/DurationInput.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    const durationInputValue = event.target.value.match(/\d*:?[0-5]?\d?/)
+    !meta.touched &&
```
Comment (rlecellier) [inline pro/src/ui-kit/form/DurationInput/DurationInput.tsx:35]: /!\ Le champs n'est plus éditable si on blur puis qu'on re-rentre dedans.

File: pro/src/ui-kit/form/DurationInput/utils/parseMinutesToHours.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export const parseMinutesToHours = (durationTime: number) => {
```
Comment (rlecellier) [inline pro/src/ui-kit/form/DurationInput/utils/parseMinutesToHours.ts:1]: on à `date-fns` d'installé sur le projet. 
Tu peut utiliser: 
```js 
import { minutesToHours } from 'date-fns'
```

File: pro/src/ui-kit/form/DurationInput/DurationInput.tsx (hunk 6 above)
Comment (rlecellier) [reply pro/src/ui-kit/form/DurationInput/DurationInput.tsx:35]: * click sur le champs durée
* entre une valeur: "10:10" par example
* click ailleurs sur la page
* click sur le champ durée
* entre une autre valeur "12:12" par example 
-> impossible d'ecrir, car `meta.touched === true` le `setValue` n'est jamais appelé.

File: pro/src/ui-kit/form/DurationInput/DurationInput.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      durationInputValue &&
+      helpers.setValue(durationInputValue[0])
+  }
```
Comment (rlecellier) [inline pro/src/ui-kit/form/DurationInput/DurationInput.tsx:38]: verifier que le tableau `durationInputValue` contient au moins une valeurs devrait satisfaire TypeScript :)
```suggestion
    const durationInputValue = event.target.value.match(/\d*:?[0-5]?\d?/)
    durationInputValue && 
      durationInputValue.length > 0 &&
      helpers.setValue(durationInputValue[0])
  }
  }
```
